src/Infrastructure/Setting/ProjectSettingsInitializer/ProjectSettingsInitializerProcessor.php

```diff
@@ -0,0 +1,189 @@
+        $this->settingInitializerRegistry = $settingInitializerRegistry;
+        $this->projectFilesInitializer = $projectFilesInitializer;
+    }
+
+    /**
+     * @param array<\SprykerSdk\SdkContracts\Entity\SettingInterface> $settings
+     * @param \SprykerSdk\Sdk\Core\Application\Dto\ProjectSettingsInitDto $projectSettingsDto
+     *
+     * @return array<\SprykerSdk\SdkContracts\Entity\SettingInterface>
+     */
+    public function initialize(array $settings, ProjectSettingsInitDto $projectSettingsDto): array
+    {
+        $settingsToSave = $this->handleSettingsValues($settings, $projectSettingsDto);
+        $this->initializeSettings($settings);
+        $this->projectFilesInitializer->initProjectFiles();
+
+        return $settingsToSave;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isInitialized(): bool
+    {
+        return $this->projectFilesInitializer->isProjectSettingsInitialised();
+    }
+
+    /**
+     * @param array<\SprykerSdk\SdkContracts\Entity\SettingInterface> $settings
+     * @param \SprykerSdk\Sdk\Core\Application\Dto\ProjectSettingsInitDto $projectSettingsDto
+     *
+     * @return array<\SprykerSdk\SdkContracts\Entity\SettingInterface>
+     */
+    protected function handleSettingsValues(array $settings, ProjectSettingsInitDto $projectSettingsDto): array
+    {
+        $settingsToSave = [];
+
+        foreach ($settings as $setting) {
+            $settingToSave = $this->handleProjectSetting($setting, $projectSettingsDto);
+
+            if ($settingToSave === null) {
+                continue;
+            }
+
+            $settingsToSave[] = $settingToSave;
+        }
+
+        return $settingsToSave;
+    }
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\SettingInterface $setting
+     * @param \SprykerSdk\Sdk\Core\Application\Dto\ProjectSettingsInitDto $projectSettingsDto
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\SettingInterface|null
+     */
+    protected function handleProjectSetting(SettingInterface $setting, ProjectSettingsInitDto $projectSettingsDto): ?SettingInterface
+    {
+        $values = $projectSettingsDto->getSettingValue($setting->getPath());
+
+        if ($projectSettingsDto->useDefaultValue() || $this->isInitializedByInitializer($setting)) {
+            return null;
+        }
+
+        if ($this->isValuesEmpty($values)) {
+            if (!$this->changeDefaultValueQuestion->ask($setting)) {
+                return null;
+            }
+
+            $values = $this->settingValueQuestion->ask($setting, $setting->getValues());
+        }
+
+        if (!$this->isValuesChanged($setting, $values)) {
+            return null;
+        }
+
+        $setting->setValues($values);
+
+        return $setting;
+    }
+
+    /**
+     * @param mixed $values
+     *
+     * @return bool
+     */
+    protected function isValuesEmpty($values): bool
+    {
+        return $values === null || $values === [];
```

Comment:
what if value will be `''`? Isn't it better to check `!$value` instead?

Reply:
`' '` is a valid values as well as `false` , `0`